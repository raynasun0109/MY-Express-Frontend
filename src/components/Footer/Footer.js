import './Footer.scss';
import logo from '../../assets/MY-Shopaholic-logo.png';
import Copyright from '../Copyright/Copyright';
import InstagramIcon from '@mui/icons-material/Instagram';
import LinkedInIcon from '@mui/icons-material/LinkedIn';
import FacebookIcon from '@mui/icons-material/Facebook';

function Footer(){



    return (
        <div className="footer_container" >
          <div className="footer_container_top">
            <div className="footer_content_container">
                <div className="footer_content_container_img">
                    <img src={logo} alt={logo} />
                </div>
                <div className="footer_content_container_text">
                  Best shopping experience forever
              </div>
            </div>
            <div className="footer_content_container">
                <div className="footer_content_container_content_title">Company</div>
                <ul className="list">
                  <li>
                  <a href={`http://${window.location.host}/about_us`}>
                      <i className="fa fa-angle-right" aria-hidden="true" />
                      {' '}
                      About us
                  </a>
                  </li>
                  <li>
                    <a href="/contact-us">
                      <i
                        className="fa fa-angle-right"
                        aria-hidden="true"
                      />
                      {' '}
                      Contact us
                    </a>
                  </li>
                </ul>
              </div>
              <div className="footer_content_container">
              <div className="footer_content_container_content_title">Policy</div>
                <ul className="list">
                  <li>
                    <a href="#">
                      <i className="fa fa-angle-right" aria-hidden="true" />
                      {' '}
                      Support
                    </a>
                  </li>
                  <li>
                    <a href="/privacy">
                      <i className="fa fa-angle-right" aria-hidden="true" />
                      {' '}
                      Privacy Policy
                    </a>
                  </li>
                </ul>
              </div>
              <div className="footer_content_container">
                <div className="footer_content_container_content_title">Follow Us</div>
                <table className="follow_us_table">
                <tbody>
                  <tr>
                    <td>
                      <a href="#">
                        <InstagramIcon/>
                      </a>
                    </td>
                    <td>
                      <a href="#">
                        <LinkedInIcon/>
                      </a>
                    </td>
                    <td>
                      <a href="#">
                        <FacebookIcon/>
                      </a>
                    </td>
                  </tr>
                  </tbody>
                </table>
            </div>
          </div>
          <div>
           <Copyright/>
          </div>
        </div>


    )
}

export default Footer;