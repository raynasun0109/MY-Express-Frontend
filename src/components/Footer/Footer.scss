@import './../../config/_styles.scss';

.footer_container{
    background-color: $base_color;
    padding:50px $layout_horizontal 20px $layout_horizontal;
    &_top{
        display: flex;
        justify-content: space-evenly;

    
    }

    .footer_content_container{
        width: 100%;
        &_img {
            margin-bottom: 20px;
            max-width: 100px;
            img{
                width: 100%;
                height: 100%;
            }
        }
        &_text{
            @extend .middle_font;
            color: $red;
            font-weight:$heavy_bold;
            max-width: 50%;
        }
        ul{
            padding-left: 0px;
             li{
                 margin:10px 0;
                list-style-type: none;
                 a{
                    @extend .middle_font;
                     text-decoration: none;
                     color:$dark_grey;
                     &:hover{
                        cursor: pointer;
                     }
                 }
             }
         }
        .follow_us_table{
            margin:1em 0;
            tr{
                td{
                    margin-right: 15px;
                    a{
                        svg{
                            path{
                                fill:$dark_grey;
                            }
                        }
                    }
                }
            }
         }
        &_content_title{            
                @extend .middle_font;
                color: $red;
                font-weight:$heavy_bold;
        }
    }

}