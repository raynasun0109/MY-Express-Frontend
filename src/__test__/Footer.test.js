import renderer from 'react-test-renderer';
import Footer from '../components/Footer/Footer.js';

it('Renders footer correctly', () => {
    const tree = renderer
      .create(<Footer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });