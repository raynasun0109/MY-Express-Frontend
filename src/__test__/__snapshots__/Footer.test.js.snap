// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders footer correctly 1`] = `
<div
  className="footer_container"
>
  <div
    className="footer_container_top"
  >
    <div
      className="footer_content_container"
    >
      <div
        className="footer_content_container_img"
      >
        <img
          alt="MY-Shopaholic-logo.png"
          src="MY-Shopaholic-logo.png"
        />
      </div>
      <div
        className="footer_content_container_text"
      >
        Best shopping experience forever
      </div>
    </div>
    <div
      className="footer_content_container"
    >
      <div
        className="footer_content_container_content_title"
      >
        Company
      </div>
      <ul
        className="list"
      >
        <li>
          <a
            href="http://localhost/about_us"
          >
            <i
              aria-hidden="true"
              className="fa fa-angle-right"
            />
             
            About us
          </a>
        </li>
        <li>
          <a
            href="/contact-us"
          >
            <i
              aria-hidden="true"
              className="fa fa-angle-right"
            />
             
            Contact us
          </a>
        </li>
      </ul>
    </div>
    <div
      className="footer_content_container"
    >
      <div
        className="footer_content_container_content_title"
      >
        Policy
      </div>
      <ul
        className="list"
      >
        <li>
          <a
            href="#"
          >
            <i
              aria-hidden="true"
              className="fa fa-angle-right"
            />
             
            Support
          </a>
        </li>
        <li>
          <a
            href="/privacy"
          >
            <i
              aria-hidden="true"
              className="fa fa-angle-right"
            />
             
            Privacy Policy
          </a>
        </li>
      </ul>
    </div>
    <div
      className="footer_content_container"
    >
      <div
        className="footer_content_container_content_title"
      >
        Follow Us
      </div>
      <table
        className="follow_us_table"
      >
        <tbody>
          <tr>
            <td>
              <a
                href="#"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="InstagramIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
                  />
                </svg>
              </a>
            </td>
            <td>
              <a
                href="#"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LinkedInIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                  />
                </svg>
              </a>
            </td>
            <td>
              <a
                href="#"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="FacebookIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
                  />
                </svg>
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
    >
      Copyright Â© 
      MY Shopaholic
       
      2023
      .
    </p>
  </div>
</div>
`;
